### PR TITLE
Fix prediction-markets tests, I

### PR DESCRIPTION
**Pull request by maltekliemann**
### What does it do?

- Tests `admin_move_market_to_resolved` with disputed markets.
- Removes test `it_resolves_a_disputed_market` involving `simple_disputes`.
- Moves `it_resolves_a_disputed_court_market` to integration tests.
- Uses more granular origins for testing.
- Fixes wrong use of origins in benchmarks.
- Adds missing tests for `buy_complete_set`.
- Adds missing tests for `create_market`.

### Is there something left for follow-up PRs?

Remove all other TODOs. Doing that in separate PRs to avoid bloating this one.

### CODE REVIEW

**Review by sea212 — DISMISSED** [review]

File: zrml/prediction-markets/src/benchmarks.rs [hunk 1]
```diff
@@ -456,7 +456,7 @@ benchmarks! {
             Some(MarketDisputeMechanism::Court),
         )?;
 
-        let approve_origin = T::ApproveOrigin::try_successful_origin().unwrap();
+        let approve_origin = T::RequestEditOrigin::try_successful_origin().unwrap();
```
[the remark this answers is not in the document]
Comment (maltekliemann) [reply zrml/prediction-markets/src/benchmarks.rs]: [ea06c5a](https://github.com/zeitgeistpm/zeitgeist/pull/1265/commits/ea06c5a79faeda875718173fcccd89b3bd00004e)

[the remark this answers is not in the document]
Comment (sea212) [reply zrml/prediction-markets/src/benchmarks.rs]: Most of the variable names in https://github.com/zeitgeistpm/zeitgeist/pull/1265/files#diff-184a7392ce5f21eba8ab19ba927496f2f752e2977acb4e49ea66b3c454f090a5 are subject to this.

File: zrml/prediction-markets/src/benchmarks.rs [hunk 2]
```diff
@@ -280,7 +280,7 @@ benchmarks! {
             ).unwrap();
         }
 
-        let close_origin = T::CloseOrigin::try_successful_origin().unwrap();
+        let close_origin = T::ResolveOrigin::try_successful_origin().unwrap();
         let call = Call::<T>::admin_move_market_to_resolved { market_id };
     }: {
         call.dispatch_bypass_filter(close_origin)?
```
Comment (sea212) [inline zrml/prediction-markets/src/benchmarks.rs:286]:
```suggestion
        let resolve_origin = T::ResolveOrigin::try_successful_origin().unwrap();
        let call = Call::<T>::admin_move_market_to_resolved { market_id };
    }: {
        call.dispatch_bypass_filter(resolve_origin)?
```

File: zrml/prediction-markets/src/benchmarks.rs [hunk 3]
```diff
@@ -316,7 +316,7 @@ benchmarks! {
...
```
Comment (sea212) [inline zrml/prediction-markets/src/benchmarks.rs:322]:
```suggestion
        let resolve_origin = T::ResolveOrigin::try_successful_origin().unwrap();
        let call = Call::<T>::admin_move_market_to_resolved { market_id };
    }: {
        call.dispatch_bypass_filter(resolve_origin)?
```

File: zrml/prediction-markets/src/benchmarks.rs [hunk 4]
```diff
@@ -368,7 +368,7 @@ benchmarks! {
...
```
Comment (sea212) [inline zrml/prediction-markets/src/benchmarks.rs:374]:
```suggestion
        let resolve_origin = T::ResolveOrigin::try_successful_origin().unwrap();
        let call = Call::<T>::admin_move_market_to_resolved { market_id };
    }: {
        call.dispatch_bypass_filter(resolve_origin)?
```

File: zrml/prediction-markets/src/benchmarks.rs [hunk 5]
```diff
@@ -421,7 +421,7 @@ benchmarks! {
...
```
Comment (sea212) [inline zrml/prediction-markets/src/benchmarks.rs:427]:
```suggestion
        let resolve_origin = T::ResolveOrigin::try_successful_origin().unwrap();
        let call = Call::<T>::admin_move_market_to_resolved { market_id };
    }: {
        call.dispatch_bypass_filter(resolve_origin)?
```

File: zrml/prediction-markets/src/benchmarks.rs [hunk 6]
```diff
@@ -529,7 +529,7 @@ benchmarks! {
         .dispatch_bypass_filter(RawOrigin::Signed(caller.clone()).into())?;
         let market_id = zrml_market_commons::Pallet::<T>::latest_market_id()?;
 
-        let approve_origin = T::ApproveOrigin::try_successful_origin().unwrap();
+        let approve_origin = T::RequestEditOrigin::try_successful_origin().unwrap();
         let edit_reason = vec![0_u8; 1024];
         Call::<T>::request_edit{ market_id, edit_reason }
         .dispatch_bypass_filter(approve_origin)?;
```
Comment (sea212) [inline zrml/prediction-markets/src/benchmarks.rs:535]:
```suggestion
        let request_edit_origin = T::RequestEditOrigin::try_successful_origin().unwrap();
        let edit_reason = vec![0_u8; 1024];
        Call::<T>::request_edit{ market_id, edit_reason }
        .dispatch_bypass_filter(request_edit_origin)?;
```

File: zrml/prediction-markets/src/benchmarks.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (sea212) [reply zrml/prediction-markets/src/benchmarks.rs]: I added another review down below that suggests all the necessary changes. You can just batch-commit them if you like via the review page.

[the remark this answers is not in the document]
Comment (maltekliemann) [reply zrml/prediction-markets/src/benchmarks.rs]: My bad, didn't double-check if anything else was wrong. 👍

**Review by sea212 — APPROVED** [review]

**Pull request merged**